go/types/map_test.go

```diff
@@ -181,12 +182,22 @@ func newMapTestSuite(size uint, expectRefStr string, expectChunkCount int, expec
 			expectPrependChunkDiff: expectPrependChunkDiff,
 			expectAppendChunkDiff:  expectAppendChunkDiff,
 			validate: func(v2 Collection) bool {
+				if v2.Len() != uint64(elems.entries.Len()) {
```

Comment:
So...This test fails on ToT, but very infrequently. I've added some additional logging code that shows exactly what's failing, and it's really confusing. As far as I can tell, two Structs with the same Type, contents and Hash fail the equality check. I have no idea how that's possible, though I assume a deeply insidious race :-/